sdk/core/allocator/alloc.h

```diff
@@ -423,53 +680,63 @@ class __packed __aligned(MALLOC_ALIGNMENT) MChunk
 	}
 #endif
 
-	// Write the revocation epoch into the header.
-	void epoch_write()
+	/**
+	 * Erase the MChunk-specific metadata of this chunk (specifically, ring)
+	 * without changing its header.
+	 */
+	void metadata_clear()
 	{
-		epoch_enq = revoker.system_epoch_get();
+		static_assert(sizeof(*this) == sizeof(MChunkHeader) + sizeof(ring));
+		/*
+		 * This is spelled slightly oddly as using memset results in a call
+		 * rather than a single store instruction.
+		 */
+		static_assert(sizeof(ring) == sizeof(uintptr_t));
+		*reinterpret_cast<uintptr_t *>(&this->ring) = 0;
```

Comment:
Does `capaligned_zero()` help? I wonder if the compiler is clever enough to collapse it into a single cap store.

Reply:
No, that also generates a call, AFAICT.